concordium-node/src/p2p/peers.rs

```diff
@@ -164,9 +164,10 @@ fn calculate_average_throughput(
         "Time went backwards or did not change. Refusing to calculate average throughput."
     );
     let delta: u64 = (now_millis - before_millis) as u64; // as is safe since we checked the difference is positive.
-
-    let avg_bps_in = (milliseconds_to_second * (bytes_recv - prev_bytes_recv)) / delta;
-    let avg_bps_out = (milliseconds_to_second * (bytes_sent - prev_bytes_sent)) / delta;
+    let recv_diff = if bytes_recv > prev_bytes_recv { bytes_recv - prev_bytes_recv } else {0};
```

Comment:
I think it would be better to make use of the fact that `calculate_average_throughput` already has return type `anyhow::Result<(u64, u64)>`. 

Hence instead of _defaulting_ to 0 as you propose, I think it would be better to ensure the provided arguments are OK with something like:
```suggestion
ensure!(
        bytes_recv > prev_bytes_recv,
        "Received bytes was lost. Refusing to calculate average throughput."
    );
    let avg_bps_in = (milliseconds_to_second * (bytes_recv - prev_bytes_recv)) / delta;

    ensure!(
        bytes_sent > prev_bytes_sent,
        "Sent bytes was lost. Refusing to calculate average throughput."
    );
    let avg_bps_out = (milliseconds_to_second * (bytes_sent - prev_bytes_sent)) / delta;
```

Also it could be nice to extend the test below with something along the lines:
       
 ```
assert!(
    calculate_average_throughput(1, 1001, 1001, 1, 1001, 1).is_err(),
    "Received bytes was lost. Refusing to calculate average throughput."
);
assert!(
    calculate_average_throughput(1, 1001, 1, 1001, 1001, 1).is_err(),
    "Sent bytes was lost. Refusing to calculate average throughput."
);
```